src/Traverser.php

```diff
@@ -2,198 +2,208 @@
 
 namespace WebHappens\Prismic;
 
+use WebHappens\Prismic\Document;
 use Illuminate\Support\Collection;
 
 class Traverser
 {
+    public static $defaultRelations = [
+        'id' => 'id',
+        'parent' => 'parent',
+        'children' => 'children',
+    ];
+
     protected $query;
     protected $document;
-    protected $parentId;
-    protected $childrenIds;
-    protected $parentMethods;
-    protected $childrenMethods;
+    protected $relations = [];
+
+    public static function make(...$parameters)
+    {
+        return new static(...$parameters);
+    }
 
-    public function __construct($document)
+    public function __construct(Document $document, $relations = [])
     {
         $this->document = $document;
 
         $this->query = Query::eagerLoadAll();
```

Comment:
Will this be abstracted in the future if we want to make the class more reusable?

Reply:
Yeah, I think it won't be based on Document specifically, and also you'll pass in some kind of repository with an interface so it can find things - really only need for the find methods though I think.